metabase/metabase
# Build datetime filters as datetime before question is saved

In this PR we check custom expression clauses for their type, and if they are `datetime`, our expression typeinferencer will reflect that.

This is used for displaying custom filter options in the visual query builder, when we are creating an unsaved question.

### How to Test

👀 This is to be tested **before saving the question**.

1. `+ New`
2. `Question`
3. `Sample database`
4. `Orders`
5. Add custom column, something like `datetimeAdd("2022-01-01", 1, "month")` and variations thereof.
6. Call custom column `Example`, we will refer to this name below
7. `Add filters to narrow your answer`

Filter option `Example` should display a `calendar` icon.

<img width="671" alt="image" src="https://user-images.githubusercontent.com/380816/204594749-6fa9e681-6ab4-4180-9898-55747b753a8e.png">

Clicking on that filter should display time-related options.

<img width="540" alt="image" src="https://user-images.githubusercontent.com/380816/204594905-c1e049bd-b508-4df5-9802-a9477bf069fa.png">
